lightning_client.go

```diff
@@ -806,6 +811,80 @@ type NetworkInfo struct {
 	NumZombieChans uint64
 }
 
+// QueryRoutesRequest is the request of a QueryRoutes call.
+type QueryRoutesRequest struct {
+	// Source is the optional source vertex of the route.
+	Source *route.Vertex
+
+	// PubKey is the destination vertex.
+	PubKey route.Vertex
+
+	// LastHop is the optional last hop befor the destination.
+	LastHop *route.Vertex
+
+	// RouteHints represents the different routing hints that can be used to
+	// assist the router. These hints will act as optional intermediate hops
+	// along the route.
+	RouteHints [][]zpay32.HopHint
```

Comment:
What does "optional intermediate hops" mean? I think we only supply hints for the final hop? 
Nitty though, I guess in theory we could have hints for other private channels that don't belong to the last hop?

Reply:
Hop hints are used to simply extend the graph and while the router knows about them, inclusion in the final route is not guaranteed.